examples/next/src/components/core/arbitrary-call.tsx

```diff
@@ -0,0 +1,224 @@
+    initializing: state.initializing,
+  }));
+
+  const { account, caller, skipBalance, updateAccount } = useConfigStore(
+    (state) => ({
+      // Get the current account that is targeted by the call
+      account: state.account,
+      // Get the current address to impersonate as the caller
+      caller: state.caller,
+      // Whether to skip the native balance check when making a call
+      skipBalance: state.skipBalance,
+      // Update the account after a call
+      updateAccount: state.updateAccount,
+    }),
+  );
+
+  // Save a tx in the history and set the current processing state
+  const { setProcessing } = useTxStore((state) => ({
+    saveTx: state.saveTx,
+    setProcessing: state.setProcessing,
+  }));
+
+  // Whether each argument is valid
+  const isValid = useMemo(() => {
+    return {
+      data: isHex(dataInput) || dataInput === '',
+      value: !isNaN(Number(valueInput)),
+    };
+  }, [dataInput, valueInput]);
+
+  /* -------------------------------- HANDLERS -------------------------------- */
+  // Perform the arbitrary call with the specified parameters
+  // This will use the Tevm client to attempt to execute the call
+  // The user will be notified of any errors or the result of the call
+  const handleArbitraryCall = async () => {
+    if (!client || !account) {
+      toast.info(
+        'There was an issue retrieving provider data. Please refresh the page.',
+      );
+      return;
+    }
+    // Check if the inputs are valid, although this should not be possible
+    if (!isValid.data || !isValid.value) return;
+
+    // Display loading state
+    setCalling(true);
+    setProcessing('arbitrary-call');
+    const loading = toast.loading('Processing the call...', {
+      description: `Calling ${account.address} with the provided data and value`,
+    });
+
+    // Process the transaction
+    const tx = await client.call({
+      from: caller,
+      to: account.address,
+      data: isHex(dataInput) ? dataInput : '0x',
+      value: valueInput === '' ? BigInt(0) : BigInt(valueInput),
+      skipBalance,
+    });
+
+    // Report the result of the transaction to the user
+    if (tx.errors?.length) {
+      toast.error(tx.errors[0].name, {
+        id: loading,
+        description: tx.errors[0].message,
+      });
+    } else {
+      const adaptedData =
+        // Show the first 100 characters of the data if it's too long
+        tx.rawData.toString().length > 100
+          ? `${tx.rawData.toString().slice(0, 100)}...`
+          : // or the entire data if it's less than 100 characters
+            tx.rawData;
+
+      toast.success('Transaction successful!', {
+        id: loading,
+        description:
+          // Don't try to display if there was no return value/transaction reverted/failed to retrieve it
+          tx.rawData === '0x'
+            ? 'See below for more details.'
+            : `Returned: ${adaptedData}. See below for more details.`,
+      });
+    }
```

Comment:
Consider pulling the toasting logic out into it's own utilitie that takes a `CallResponse` as a param